src/base/preferences.cpp

```diff
@@ -1330,6 +1330,19 @@ void Preferences::setMarkOfTheWebEnabled(const bool enabled)
     setValue(u"Preferences/Advanced/markOfTheWeb"_s, enabled);
 }
 
+bool Preferences::isValidateTLSCertificate() const
+{
+    return value(u"Preferences/Advanced/ValidateTLSCertificate"_s, true);
+}
+
+void Preferences::setValidateTLSCertificate(bool enabled)
+{
+    if (enabled == isValidateTLSCertificate())
```

Comment:
Here I followed the convention of the surrounding code. However, I am questioning the need for this check.
`setValue()` and its descendants eventually check if the value has changed before writing to disk and before emiting a signal.

Reply:
>However, I am questioning the need for this check.

It can help skip acquiring the QWriteLocker in `SettingsStorage`.